src/Analyzers/CSharp/CodeFixes/ConvertNameOf/CSharpConvertNameOfCodeFixProvider.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+               context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var isUsingSystem = root.
+                                DescendantNodes().
+                                OfType<UsingDirectiveSyntax>().
+                                Any(node => node.Name.ToString().Equals("System"));
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = (MemberAccessExpressionSyntax)root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                ConvertNameOf(editor, node, semanticModel, isUsingSystem);
+            }
+        }
+
+        /**
+         * Method converts typeof(...).Name to nameof(...)
+         * The isUsingSystem parameter determines whether includes the System directive.
+         */
+        internal static void ConvertNameOf(
+            SyntaxEditor editor, MemberAccessExpressionSyntax node, SemanticModel semanticModel, bool isUsingSystem)
+        {
+            var exp = (TypeOfExpressionSyntax)node.Expression;
+            var idName = exp.Type.ToString();
+
+            //check if exp type is predefined type and convert
+            //example: int -> Int32, string -> String, etc
+            if (exp.Type.IsKind(SyntaxKind.PredefinedType))
+            {
+                idName = semanticModel.GetSymbolInfo(exp.Type).Symbol.
+                                       GetSymbolType().SpecialType.
+                                       ToPredefinedType().ToString();
+            }
+
+            //check if user is using System;
+            idName = isUsingSystem ? idName : "System." + idName;
+
+            var nameOfSyntax = InvocationExpression(IdentifierName("nameof")).
+                               WithArgumentList(
+                               ArgumentList(
+                               SingletonSeparatedList<ArgumentSyntax>(
+                               Argument(
+                               IdentifierName(idName)))));
+
+            editor.ReplaceNode(node, nameOfSyntax.WithAdditionalAnnotations(Simplifier.Annotation));
+        }
+
+        private class MyCodeAction : CustomCodeActions.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(CSharpAnalyzersResources.Convert_type_name_to_nameof, createChangedDocument)
```

Comment:
```suggestion
                : base(CSharpAnalyzersResources.Convert_type_name_to_nameof, createChangedDocument, CSharpAnalyzersResources.Convert_type_name_to_nameof)
```

Reply:
add fix-all tests.

Reply:
basic fix all tests have been added, more to come